Enablers for fixing C2 accelerometer / orientation problems

spiiroin opened the pull request.

spiiroin committed:
[hybrisadapter] Implement doubleStopReader quirk. JB#62669

In C2 disabling gyroscope when also accelerometer is enabled can lead
to accelerometer reporting to freeze. Sensor drivers are not available
for analysis / fixing, but it seems that forcing gyroscope restart in
different datarate resumes accelerometer reporting.

As this is a workaround for a device specific problem, this behavior
is disabled by default and should be enabled on as-needed basis in
device specific sensorfwd configuration.

Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>

spiiroin committed:
[orientation] Use milliG values for accelerometer limits. JB#62669

Orientation interpreter ignores accelerometer values that are out of
bounds. These limits can be set in configuration files using somewhat
curious [milliG²/1000] units.

As no currently supported device uses non-default values, we can
safely switch the limits to be defined in milliG units.

Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>

pvuorela commented on core/hybrisadaptor.cpp:
```diff
@@ -882,10 +892,27 @@ void HybrisManager::startReader(HybrisAdaptor *adaptor)
-            }
+        sensordLogD() << "deactivating " << adaptor->name();
+        if (!setActive(adaptor->m_sensorHandle, false)) {
+            sensordLogW() <<Q_FUNC_INFO<< "failed";
```
Old code but could have spaces around Q_FUNC_INFO.

As side note, these logging functions seem like pointless wrappers for qt. Could switch to qt category logging, but that's for different PR.

spiiroin replied:
spaces added

direc85 commented on filters/orientationinterpreter/orientationinterpreter.cpp:
```diff
@@ -120,8 +120,8 @@ void OrientationInterpreter::accDataAvailable(unsigned, const AccelerationData*
 {
-    int vector = ((data.x_ * data.x_ + data.y_ * data.y_ + data.z_ * data.z_) / 1000);
-    return !((vector >= minLimit) && (vector <= maxLimit));
+    int m = int(sqrtf(data.x_ * data.x_ + data.y_ * data.y_ + data.z_ * data.z_));
+    return m < minLimit || m > maxLimit;
```
Would be faster to drop `sqrtf()` and compare against (cached) squared limits?

spiiroin replied:
hmm, that might make sense - keep the good part of old logic

spiiroin replied:
Done as suggested + tried to lessen chances of numerical overflows causing problems.

pvuorela commented on core/hybrisadaptor.cpp:
```diff
@@ -298,6 +299,15 @@ bool HybrisManager::typeRequiresWakeup(int type)
 void HybrisManager::initManager()
 {
+    QString sensorTypes = SensorFrameworkConfig::configuration()->value("hybrisQuirks/doubleStopReader", QString());
+    for (const QString &iter : sensorTypes.split(" ", QString::SkipEmptyParts)) {
```
This will break Qt6 build. Could do the same thing we have here around with
#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)

spiiroin replied:
hmpf. Or, since empty slots are unlikely to produce > 0 int values, might as well skip the SkipEmptyParts

spiiroin committed:
[config] Stop QSettings from using backing file. Fixes JB#56894

Sensorfwd uses QSettings for reading configuration files, which
implicitly makes sensorfwd also read and write configuration data
in (not properly set up) standard location like
"/root/.config/Unknown Organization/sensorfwd.conf".

Which then causes sensorfwd to "remember" content of configuration
files that once existed in but that have since been removed from
"/etc/sensorfw/sensord.conf.d" directory.

Switch from default QSettings constructor to no-backing-file one so
that default application directories are not read from / written to.

Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>

pvuorela commented:
Still good
